src/js/directives/image-list.directive.js

```diff
@@ -30,17 +30,31 @@
         $log.error('Images List Failed! Please provide an array.');
 
       } else {
+        _generateListItems();
 
-        for (var i = 0; i < controller.images.length; i++) {
+      }
+
+      scope.$watch(function(){
+        return controller.images;
+      }, function(newVal, oldVal){
+
+        if(newVal !== oldVal){
+          _generateListItems();
+        }
+
+      });
+
+      function _generateListItems(){
+
+        angular.forEach(controller.images, function (value) {
```

Comment:
This line isn't wrong, just a preference for readability. I typically label my single "item" with something referencing the item itself. So, in this case we're looping over `images`, so I'd name my item `image`. If we were looping over `users`, I'd name it `user`. This clearly represents what the object is to the next dev reading the code.